CSharpGeneratorFunctions.sh

```diff
@@ -1,18 +1,28 @@
 # This is intended to be imported using the "source" function
 # from any script that wants to invoke the C#-based generator.
 
-CSHARP_GENERATOR_DIR=tmp-gapic-generator-csharp
+TMP_CSHARP_GENERATOR_DIR=tmp-gapic-generator-csharp
+if [[ $CSHARP_GENERATOR_DIR == "" ]]
```

Comment:
+1 to this. I always end up modifying the script and then reverting :)

Reply:
Ditto!